pkg/controller/storagecluster/stork.go

```diff
@@ -783,13 +850,20 @@ func (c *Controller) createStorkSchedDeployment(
 		kubeSchedImage = "k8s.gcr.io/kube-scheduler-amd64"
 	}
 
-	k8sVersion, err := version.NewVersion(minK8SVersionForPinnedStorkScheduler)
+	k8sMinVersionForPinnedStorkScheduler, err := version.NewVersion(minK8sVersionForPinnedStorkScheduler)
 	if err != nil {
-		logrus.WithError(err).Errorf("Could not parse version %s", minK8SVersionForPinnedStorkScheduler)
+		logrus.WithError(err).Errorf("Could not parse version %s", k8sMinVersionForPinnedStorkScheduler)
 		return err
 	}
 
-	if c.kubernetesVersion.GreaterThanOrEqual(k8sVersion) {
+	k8sMinVersionForKubeSchedulerConfiguration, err := version.NewVersion(minK8sVersionForKubeSchedulerConfiguration)
+	if err != nil {
+		logrus.WithError(err).Errorf("Could not parse version %s", k8sMinVersionForKubeSchedulerConfiguration)
+		return err
+	}
+
+	if c.kubernetesVersion.GreaterThanOrEqual(k8sMinVersionForPinnedStorkScheduler) &&
+		!c.kubernetesVersion.GreaterThanOrEqual(k8sMinVersionForKubeSchedulerConfiguration) {
```

Comment:
For backward compatibility, let's start using the kube-scheduler configuration from 1.25.

Users who are using air-gapped or custom registry, may not have the correct image uploaded to their registries. If they update to this version of the operator, the stork-scheduler will fail to come up because now we will start looking for a different stork-scheduler image. It will break the operator upgrade workflow for existing customers.

Reply:
Beazleay reported this issue and they are on K8 1.23.15. They are waiting for this fix in Operator.  If not using Kube Scheduler Configuration, Stork will not spread pod uniformly on K8 1.23 and 1.24. So it's going to be a problem for the customers who will continue to be on the previous version.  Or is it that we will force the customers to upgrade the K8 version prior to moving to the next operator version?

Reply:
Opened a [PD ticket](https://portworx.atlassian.net/browse/PD-1752) Will get this info updated in the Operator upgrade section for air gapped cluster.